### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by jonrandy**
#### 1. Objective
- Fix the 500 error occurring when payments fail on Prestashop 1.6
- Fix the English on the 'payment error' page
- Fix the longstanding issue with checkboxes sometimes not appearing to specifiy what currencies to allow for the Omise payment plugin
- Fix the error thrown by the Webhooks controller (issue #67)

#### 2. Description of change
- Adds a version specific template for the 'Payment error' page
- Changes the path for the webhooks template as it differs between Prestashop versions
- Changes the English message displayed on these templates
- Changes the configuration/setup of the plugin so that `need_instance` is turned on - the instance is always then available as needed on the payment plugin prefs admin page (fix gleaned from extensive searching on Google and forums)

#### 3. Quality assurance

Re-checked all affected areas on both versions of Prestashop (payment error page and payment prefs admin page)

#### 4. Impact of the change

All bugs listed in Objectives are now fixed

#### 5. Priority of change

High. Merchants waiting on fixes

#### 6. Additional notes

Do you personally accept that `1 / 0 = ∞`?

### CODE REVIEW

File: omise/controllers/front/base.php [hunk 1]
```diff
@@ -47,7 +50,7 @@
     public function __construct()
     {
         parent::__construct();
-
+
```
Comment (guzzilar) [inline omise/controllers/front/base.php]: @jonrandy there is an unnecessary space here.

File: omise/controllers/front/base.php [hunk 2]
```diff
@@ -3,6 +3,9 @@
     exit();
 }
 
+if (!defined('IS_VERSION_17')) define('IS_VERSION_17', _PS_VERSION_ >= '1.7');
+
+
```
Comment (guzzilar) [inline omise/controllers/front/base.php]: @jonrandy probably we need just only 1 breakline?

File: omise/controllers/front/base.php [hunk 3]
```diff
@@ -91,7 +94,7 @@ public function initContent()
      * @see FrontControllerCore::postProcess()
      */
     public function postProcess()
-    {
+    {
```
Comment (guzzilar) [inline omise/controllers/front/base.php]: @jonrandy there is an unnecessary space here.

File: omise/views/templates/front/payment-error16.tpl [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{capture name=path}
```
Comment (guzzilar) [inline omise/views/templates/front/payment-error16.tpl]: @jonrandy (optional) To make this file structure consistent with the one in `omise-prestashop/omise/views/templates/hook/` Do you think we should have  
```
- omise/views/templates/front/1.7/payment-error.tpl
- omise/views/templates/front/1.6/payment-error.tpl
```` 
instead?

**Review by guzzilar — APPROVED** [review]
👍 LGTM, just few more on code cosmetic stuff.

**Comment by guzzilar** [comment]
@jonrandy can you also refer to the issue this PR is fixing in the pull request description? (I guess this one #67 ?)

**Review by jacstn — APPROVED** [review]

**Pull request merged**